sos/archive.py

```diff
@@ -559,6 +559,42 @@
         self._archive_root = _new_root
         self._archive_name = os.path.join(self._tmp_dir, self.name())
 
+    def do_file_sub(self, path, regexp, subst):
+        """Apply a regexp substitution to a file in the archive.
+
+        :param path: Path in the archive where the file can be found
+        :type path: ``str``
+
+        :param regexp:  A regex to match the contents of the file
+        :type regexp: ``str`` or compiled ``re`` object
+
+        :param subst: The substitution string to be used to replace matches
+                      within the file
+        :type subst: ``str``
+
+        :returns: Number of replacements made
+        :rtype: ``int``
+        """
+        common_flags = re.IGNORECASE | re.MULTILINE
+        if hasattr(regexp, "pattern"):
+            pattern = regexp.pattern
+            flags = regexp.flags | common_flags
+        else:
+            pattern = regexp
+            flags = common_flags
+
+        readable = self.open_file(path)
```

Comment:
This should be fine; the file is closed at the end of the `do_file_sub` method when both local variables `readable` and `content` are vanished.

Meantime, we just read the content of the file once and *only then* we potentially overwrite it (when some replacement is found). No concurrent reading and writing can happen.

Reply:
That's what this code check is mainly concerned with - the file is not _guaranteed_ to be closed if for instance we hit an issue prior to the end of `do_file_sub()`.

This can be resolved by a context manager, e.g.

```python
with self.open_file(path) as readable:
   [nest everything here]
```

Reply:
OK, it makes sense. I added declaration of the `content` variable just in a (hypothetical) case the `with` block fails and the variable remains undeclared.